core/src/main/java/io/grpc/ServerCall.java

```diff
@@ -176,4 +176,16 @@ public boolean isReady() {
   public void setMessageCompression(boolean enabled) {
     // noop
   }
+
+  /**
+   * Sets the compression algorithm for this call.  If the server does not support the compression
+   * algorithm, the call will fail.
+   *
+   * @param compressor the name of the compressor to use.
+   * @throws IllegalArgumentException if the compressor name can not be found.
+   */
+  @ExperimentalApi
+  public void setCompression(String compressor) {
+    throw new UnsupportedOperationException();
```

Comment:
This doesn't seem to be a valid implementation of this method. If it were, then I would expect the JavaDoc to explicitly state when `UnsupportedOperationException` would be thrown. But I would also find it painful if this method was allowed to throw UnsupportedOperationException.

If the only concern is interceptors forwarding calls to a concrete implementation, then the fix is not this exception but to have that code use `ForwardingServerCall`.